x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcDatabaseMetaData.java

```diff
@@ -1220,4 +1229,61 @@ public void close() throws SQLException {
             // this cursor doesn't hold any resource - no need to clean up
         }
     }
+
+    static String escapeStringPattern(String pattern, EscapeWildcard strategy) {
+        if (StringUtils.hasText(pattern) == false || "%".equals(pattern)) {
+            return pattern;
+        }
+
+        char esc = '\\';
+
+        switch (strategy) {
+            case NEVER:
+                return pattern;
+            case ALWAYS:
+                return pattern.replace("_", esc + "_").replace("%", esc + "%");
+            default:
+                // first detect if there's any escaping in the string, if so bail out
+                boolean escaped = false;
+                boolean hasEscaping = false;
+
+                for (int i = 0; i < pattern.length(); i++) {
+                    char curr = pattern.charAt(i);
+                    if (escaped == false && (curr == esc) && esc != 0) {
+                        escaped = true;
+                    } else {
+                        if ((curr == '%' || curr == '_' || curr == esc) && (escaped == true)) {
+                            hasEscaping = true;
```

Comment:
Here you could just `return pattern;` since you break from the loop and then you just return.